framework/transport/ieee1905_transport/ieee1905_transport_broker.cpp

```diff
@@ -0,0 +1,355 @@
+            LOG(ERROR) << "Failed sending message with type (0x" << std::hex << msg_opcode.value
+                       << std::dec << ") to FD (" << soc->getSocketFd() << ")";
+
+            continue;
+        }
+    }
+
+    return true;
+}
+
+void BrokerServer::register_internal_message_handler(MessageHandler handler)
+{
+    LOG_IF(m_internal_message_handler, WARNING) << "Overriding previously registered handler";
+    m_internal_message_handler = handler;
+}
+
+void BrokerServer::register_external_message_handler(MessageHandler handler)
+{
+    LOG_IF(m_internal_message_handler, WARNING) << "Overriding previously registered handler";
+    m_external_message_handler = handler;
+}
+
+bool BrokerServer::handle_msg(std::shared_ptr<Socket> sd)
+{
+    // Check if the socket contains enough bytes for the header
+    if (sd->getBytesReady() < static_cast<ssize_t>(sizeof(messages::Message::Header))) {
+        // Received partial header - do nothing...
+        return true;
+    }
+
+    // Read and parse the message
+    auto message = messages::read_transport_message(*sd);
+    if (!message) {
+        return false;
+    }
+
+    // Handle the specific message
+    switch (messages::Type(message->type())) {
+    // Broker Subscribe/Unsubscribe
+    case messages::Type::SubscribeMessage: {
+        auto msg_ptr = dynamic_cast<messages::SubscribeMessage *>(message.get());
+        return handle_subscribe(sd, *msg_ptr);
+    }
+    // Outgoing CMDU messages
+    case messages::Type::CmduTxMessage: {
+        LOG(DEBUG) << "Processing external message from FD: " << sd->getSocketFd();
+        if (!m_external_message_handler) {
+            LOG(ERROR) << "External message handler not registered!";
+            return false;
+        }
+
+        return m_external_message_handler(message, *this);
+    }
+    // Internal messages
+    default: {
+        LOG(DEBUG) << "Processing internal message from FD: " << sd->getSocketFd();
+        if (!m_internal_message_handler) {
+            LOG(ERROR) << "Internal message handler not registered!";
+            return false;
+        }
+
+        return m_internal_message_handler(message, *this);
+    }
+    }
+
+    return true;
+}
+
+bool BrokerServer::handle_subscribe(std::shared_ptr<Socket> &sd,
+                                    const messages::SubscribeMessage &msg)
+{
+    // Validate the message type
+    if (msg.metadata()->type == messages::SubscribeMessage::ReqType::INVALID) {
+        LOG(ERROR) << "Invalid subscribe message type!";
+        return false;
+    }
+
+    // Make sure there's at least one message type
+    if (msg.metadata()->msg_types_num == 0) {
+        LOG(ERROR) << "Subscribe message does not contain any types!";
+        return false;
+    }
+
+    // Iterate over the message types and subscribe/unsubscribe
+    bool subscribe = msg.metadata()->type == messages::SubscribeMessage::ReqType::SUBSCRIBE;
+    std::stringstream types_stream;
```

Comment:
(nitpick) I like to show in the name what its purpose is:
```suggestion
    std::stringstream log_types;
```

Reply:
Done